fuel-vm/src/interpreter/memory.rs

```diff
@@ -581,3 +588,23 @@ pub(crate) fn write_bytes<const COUNT: usize>(
     memory[range.usizes()].copy_from_slice(&bytes);
     Ok(())
 }
+
+/// Attempt copy from slice to memory, filling zero bytes when exceeding slice boundaries.
+/// Performs overflow and memory range checks, but no ownership checks.
+pub(crate) fn copy_from_slice_zero_fill_noownerchecks<A: ToAddr, B: ToAddr>(
+    memory: &mut [u8; MEM_SIZE],
+    src: &[u8],
+    dst_addr: A,
+    src_offset: usize,
+    len: B,
+) -> Result<(), RuntimeError> {
+    let range = MemoryRange::new(dst_addr, len)?;
+
+    let src_end = range.len().min(src.len());
+    let data = src.get(src_offset..src_end).unwrap_or_default();
```

Comment:
If the size of the `src`(contact) is `1000`, `len` is `8`, and `offset` is `999`, then you will have `999..8` instead of `999..1000`.

Maybe it is better to add tests for this function to cover all cases.

Also, do we need to worry about padding and alignment in the memory here?

Reply:
Fixed and added tests in 6c5313e. This function doesn't care about padding, it must be computed before it wanted. Alignment is not a concept FuelVM uses, so I'm not sure what you're asking here.